synapse/handlers/pagination.py

```diff
@@ -381,22 +379,26 @@ async def get_messages(
                         member_event_id
                     )
                     if RoomStreamToken.parse(leave_token).topological < max_topo:
-                        source_config.from_key = str(leave_token)
+                        from_token = from_token.copy_and_replace(
+                            "room_key", leave_token
+                        )
 
                 await self.hs.get_handlers().federation_handler.maybe_backfill(
                     room_id, max_topo
                 )
 
             events, next_key = await self.store.paginate_room_events(
                 room_id=room_id,
-                from_key=source_config.from_key,
-                to_key=source_config.to_key,
-                direction=source_config.direction,
-                limit=source_config.limit,
+                from_key=from_token.room_key,
+                to_key=pagin_config.to_token.room_key
+                if pagin_config.to_token
+                else None,
```

Comment:
Embedding this into the method call makes it hard to read, maybe store `to_key` into a variable first?

Reply:
I did not realise it had done that formatting :/